src/python/tensor.py

```diff
@@ -92,6 +92,16 @@ def toarray(self):
         data = np.array(data, dtype=dt).reshape(self.tuple_shape)
         return data
 
+    def from_array(self, np_array, offset=0):
```

Comment:
rename it as copy_data_from()? and make from_array a global function (not tensor member function).
similarly for copy_data_to()  --- to_array()